src/PageLayout/useStickyPaneHeight.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import {useInView} from 'react-intersection-observer'
+
+/**
+ * Calculates the height of the sticky pane such that it always
+ * fits into the viewport even when the header or footer are visible.
+ */
+// TODO: Handle sticky header
+export function useStickyPaneHeight() {
+  // Default the height to the viewport height
+  const [height, setHeight] = React.useState('100vh')
+
+  // Create intersection observers to track the top and bottom of the content region
+  const [contentTopRef, contentTopInView, contentTopEntry] = useInView()
```

Comment:
Looks like `useInView` takes a margin in the options, so it should be easy to add `stickyTop` down the road